2022/Day7/Solve.py

```diff
@@ -0,0 +1,83 @@
+class Day_Six(object):
+  def day_six(self):
+    computer = self.parse("PuzzleInput.txt")
+    computer.calculate_size()
+    print("The answer to part one is", computer.sum_all_subdirectories_under_limit(100000))
+
+  def parse(self, puzzle_input):
+    current_directory = []
+    computer = Directory("dir /")
+    with open(puzzle_input, "r") as f:
+      for line in f:
+        if "cd /" in line:
+          continue
+        if ".." in line:
+          current_directory.pop()
+          continue
+        elif "$ cd" in line:
+          current_directory.append("dir " + line.split()[2])
+          continue
+        elif "$" not in line:
+          if "dir" in line:
+            item = Directory(line.strip())
+          else:
+            item = File(line)
+          computer.add_files(current_directory, item)
+
+    return computer
+
+
+class Directory(object):
+  name: str
+  size: int
+  contents: {}
+
+  def __init__(self, name):
+    self.name = name
+    self.contents = {}
+    self.size = 0
```

Comment:
Why are you setting the size of the directory to zero?  It's not zero?

Reply:
Because when it's first created and nothing is in it, it is zero, no?

Reply:
Okay, but as soon as you add something, it's immediately not zero, right?  See my comment below about choosing _not_ to store size as a variable and calculating it on the fly.